CMakeLists.txt

```diff
@@ -73,12 +69,47 @@ option(DISABLE_JSON_PATCH             "Disable JSON patch (RFC6902) support."
 option(NEWLOCALE_NEEDS_FREELOCALE     "Work around newlocale bugs in old FreeBSD by calling freelocale"  OFF)
 option(BUILD_APPS                     "Default to building apps" ON)
 
+if (UNIX OR MINGW OR CYGWIN OR AMIGA)
+    list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
+endif()
 
-if (UNIX OR MINGW OR CYGWIN)
+if (${CMAKE_SYSTEM_NAME} STREQUAL "AmigaOS")
+    set(AMIGA ON)
+    set(DISABLE_THREAD_LOCAL_STORAGE ON)
+    set(ENABLE_THREADING OFF)
+    set(BUILD_SHARED_LIBS OFF)
+    set(BUILD_APPS OFF)
     list(APPEND CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
```

Comment:
This line is duplicating the same `-D_GNU_SOURCE` from a few lines above, so this should be removed from this block, but the block of code that sets if (lines 72-74) need to move a bit lower (so the check for `AMIGA` works)

Reply:
Good catch. Silly me